packages/dd-trace/src/exporters/agent/writer.js

```diff
@@ -70,7 +75,20 @@ class Writer {
 
     log.debug(() => `Request to the agent: ${JSON.stringify(options)}`)
 
-    platform.request(Object.assign({ data }, options), (err, res) => {
+    platform.metrics().increment('datadog.tracer.node.exporter.agent.requests')
+
+    platform.request(Object.assign({ data }, options), (err, res, status) => {
+      if (status) {
+        platform.metrics().increment('datadog.tracer.node.exporter.agent.responses')
+        platform.metrics().increment('datadog.tracer.node.exporter.agent.responses', `status:${status}`)
+      } else {
+        platform.metrics().increment('datadog.tracer.node.exporter.agent.errors')
+
+        if (err && err.code) {
+          platform.metrics().increment('datadog.tracer.node.exporter.agent.errors.by.code', `code:${err.code}`)
```

Comment:
worth reporting as `code:unknown`, then can always report the metric so `sum:.errors.by.code` will always be the same value as `.errors`?

Reply:
I'm not a fan of doing that since there is no way to then differentiate an actual `unknown` error code from no error code.